File: library_crashlytics/src/main/java/com/dolin/crashlytics/internal/ILogger.java
package com.dolin.crashlytics.internal;

/**
 * @author #Suyghur.
 * Created on 2021/07/30
 */
public interface ILogger {

    /**
     * Log a DEBUG message.
     *
     * @param tag Used to identify the source of a log message.
     * @param msg The message you would like logged.
     */
    @SuppressWarnings("unused")
    void d(String tag, String msg);

    /**
     * Log a INFO message.
     *
     * @param tag Used to identify the source of a log message.
     * @param msg The message you would like logged.
     */
    @SuppressWarnings("unused")
    void i(String tag, String msg);


    /**
     * Log a WARN message.
     *
     * @param tag Used to identify the source of a log message.
     * @param msg The message you would like logged.
     */
    @SuppressWarnings("unused")
    void w(String tag, String msg);


    /**
     * Log a ERROR message.
     *
     * @param tag Used to identify the source of a log message.
     * @param msg The message you would like logged.
     */
    @SuppressWarnings("unused")
    void e(String tag, String msg);

}

File: library_zap/src/main/java/com/suyghur/dolin/zap/internal/ILogger.kt
package com.suyghur.dolin.zap.internal

/**
 * @author #Suyghur.
 * Created on 4/7/21
 */
interface ILogger {


    fun d(any: Any?)

    fun d(tag: String, any: Any?)

    fun i(any: Any?)

    fun i(tag: String, any: Any?)

    fun w(any: Any?)

    fun w(tag: String, any: Any?)

    fun e(any: Any?)

    fun e(tag: String, any: Any?)

    fun record(level: Int, tag: String, msg: String)

}